migrations/1601022564938-copy-parents-data-into-children-entities-and-delete-parent-users.js

```diff
@@ -0,0 +1,169 @@
+		consent: {
+			parentConsents: [
+				{
+					parentId: { type: mongoose.Schema.Types.ObjectId, ref: 'user' },
+					form: { type: String, enum: consentForm },
+					dateOfPrivacyConsent: { type: Date },
+					dateOfTermsOfUseConsent: { type: Date },
+					privacyConsent: { type: Boolean },
+					termsOfUseConsent: { type: Boolean },
+				},
+			],
+		},
+	}),
+	'users'
+);
+
+const getIds = (docs) => docs.map((doc) => doc._id);
+
+const getParentRole = async () => RoleModel.findOne({ name: 'parent' }).lean().exec();
+
+const findAllParents = (parentRole) => User.find({ roles: parentRole._id });
+
+const deleteParentUsersByIds = async (ids) => {
+	const result = await User.deleteMany({
+		_id: { $in: ids },
+	}).exec();
+	info(`${result.deletedCount} users deleted.`);
+};
+
+const getStudentsForParent = async (parent) => {
+	const students = await OldUser.find({ parents: parent._id }).lean().exec();
+	info(`${students.length} students for parent with id=${parent._id} will be updated!`);
+	return students;
+};
+
+const updateStudentsParentData = async (studentIds, parent) => {
+	return User.updateMany(
+		{ _id: { $in: [studentIds] } },
+		{
+			$set: {
+				parents: [
+					{
+						firstName: parent.firstName,
+						lastName: parent.lastName,
+						email: parent.email,
+					},
+				],
+			},
+			$unset: { 'consent.parentConsents.$[].parentId': '' },
+		}
+	);
+};
+
+const migrateParentsToStudents = async () => {
+	const parentIdsToDelete = [];
+	const parentRole = await getParentRole();
+	const cursor = findAllParents(parentRole).cursor();
+	for (let parent = await cursor.next(); parent != null; parent = await cursor.next()) {
+		const students = await getStudentsForParent(parent);
+		const studentIds = getIds(students);
+		await updateStudentsParentData(studentIds, parent);
+		parentIdsToDelete.push(parent._id);
+	}
+
+	let amountOfParentsLeftToDelete = parentIdsToDelete.length;
+
+	const limit = 500;
+	while (amountOfParentsLeftToDelete !== 0) {
+		const parentsToDelete = amountOfParentsLeftToDelete < limit ? amountOfParentsLeftToDelete : limit;
+		await deleteParentUsersByIds(parentIdsToDelete.splice(0, parentsToDelete));
+		amountOfParentsLeftToDelete -= parentsToDelete;
+	}
+
+	info(`Removing role: ${parentRole.name}`);
+	await RoleModel.deleteOne({ _id: parentRole._id });
+};
+
+const findAllStudentsWithParentData = () => User.find({ parents: { $elemMatch: { email: { $ne: null } } } });
+
+const migrateParentsFromStudents = async () => {
+	const parentRole = await RoleModel.create({ name: 'parent' });
+	const cursor = findAllStudentsWithParentData().cursor();
+	for (let student = await cursor.next(); student != null; student = await cursor.next()) {
+		const parent = await User.create({
+			firstName: student.parents[0].firstName,
+			lastName: student.parents[0].lastName,
+			email: student.parents[0].email,
+			schoolId: student.schoolId,
+			childrens: [student._id],
```

Comment:
```suggestion
			children: [student._id],
```

Reply:
The model contains childrens field which is an array so I think it should stay as it is.